Get rid of -fno-strict-overflow

We only have 1 issue when this is disabled, lets fix it and get rid of the option:

```
/home/nashif/Work/github/zephyr/samples/net/throughput_server/src/server.c: In function ‘stats’:
/home/nashif/Work/github/zephyr/samples/net/throughput_server/src/server.c:157:6: error: assuming signed overflow does not occur when assu
ming that (X + c) >= X is always true [-Werror=strict-overflow]
   if (new_print > curr) {
      ^
cc1: all warnings being treated as errors
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -351,7 +351,6 @@ include(cmake/extra_flags.cmake)
 zephyr_cc_option(-fno-asynchronous-unwind-tables)
 zephyr_cc_option(-fno-pie)
 zephyr_cc_option(-fno-pic)
-zephyr_cc_option(-fno-strict-overflow)
 
 if(CONFIG_THREAD_LOCAL_STORAGE)
 # Only support local exec TLS model at this point.

